res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,434 @@
+        outKey: "enabled",
+        inKey: "group_[Channel" + deckNumber + "]_enable",
+        input: function(channel, control, value, _status, _group) {
+            this.isPress = value !== 0;
+            engine.setValue("[EffectRack1_EffectUnit1]", "group_[Channel" + deckNumber + "]_enable", this.isPress);
+            engine.setValue("[EffectRack1_EffectUnit2]", "group_[Channel" + deckNumber + "]_enable", this.isPress);
+            this.trigger();
+        }
+    });
+
+    this.dryWetKnob = new components.Pot({
+        group: "[EffectRack1_EffectUnit" + deckNumber + "]",
+        inKey: "mix",
+        shift: function() {
+            this.inKey = "super1";
+        },
+        unshift: function() {
+            this.inKey = "mix";
+        },
+    });
+};
+NumarkScratch.EffectUnit.prototype = new components.ComponentContainer();
+
+NumarkScratch.XfaderContainer = function() {
+    this.crossfader = new components.Pot({
+        midi: [0xBF, 0x08],
+        group: "[Master]",
+        inKey: "crossfader",
+    });
+
+    this.setCurve = function(channel, control, value, _status, _group) {
+        switch (value) {
+        case 0x00: // Additive/Linear
+            engine.setValue("[Mixer Profile]", "xFaderMode", 0);
+            engine.setValue("[Mixer Profile]", "xFaderCalibration", 0.4);
+            engine.setValue("[Mixer Profile]", "xFaderCurve", 0.9);
+            break;
+        case 0x7F:  // Picnic Bench/Fast Cut
+            engine.setValue("[Mixer Profile]", "xFaderMode", 0);
+            engine.setValue("[Mixer Profile]", "xFaderCalibration", 0.9);
+            engine.setValue("[Mixer Profile]", "xFaderCurve", 7.0);
+            break;
+        }
+    };
+
+    this.xFaderReverse = function(channel, control, value, _status, _group) {
+        // 0x7F is ON, 0x00 is OFF
+        engine.setValue("[Mixer Profile]", "xFaderReverse", (value === 0x7F) ? 1 : 0);
+    };
+};
+
+NumarkScratch.XfaderContainer.prototype = new components.ComponentContainer();
+
+NumarkScratch.Deck = function(number) {
+    components.Deck.call(this, number);
+
+    const channel = number;
+
+    this.channelInputPC = new components.Button({
+        group: "[Channel" + number + "]",
+        input: function(channel, control, value, status, group) {
+            if (value === 0x00) {
+                // Set vinyl control enabled and passthrough disabled
+                engine.setValue(group, "vinylcontrol_enabled", 1);
+                engine.setValue(group, "passthrough", 0);
+            }
+        }
+    });
+
+    this.channelInputLinePhono = new components.Button({
+        group: "[Channel" + number + "]",
+        input: function(channel, control, value, status, group) {
+            if (value === 0x02) {
+                // Set vinyl control disabled and passthrough enabled
+                engine.setValue(group, "vinylcontrol_enabled", 0);
+                engine.setValue(group, "passthrough", 1);
+            }
+        }
+    });
```

Comment:
@Swiftb0y, this is the last issue I have, I cannot get the status of these switches (channelInputLinePhono & channelInputPC) to reflect in Mixxx on startup. I have to manually toggle the switch to update the Mixxx UI. Also after initial failure, if I reload the script in VS Code, Mixxx UI reflects the correct state. Can I please get your help here.
<img width="149" alt="image" src="https://github.com/mixxxdj/mixxx/assets/24758479/2c865888-3f1d-42f5-b848-d40dad9fc08c">

Reply:
This is probably an order-of-initialization issue within mixxx. I dont know when ill have the time to look into it.

Reply:
no worries, whenever you get time is fine.